flow/util/genRuleFile.py

```diff
@@ -1,220 +1,295 @@
 #!/usr/bin/env python3
 
-import sys
+from math import ceil, isinf
 from os import chdir
 from os.path import isfile, abspath
 from re import sub
+import argparse
 import json
-from math import ceil, isinf
+import operator
+import sys
+
+parser = argparse.ArgumentParser(
+    description='Generates or updates rules file for CI.')
+parser.add_argument('dir', help='Path to the design directory.')
+parser.add_argument(
+    '-v',
+    '--variant',
+    default='base',
+    help='Flow variant [default="base"].')
+parser.add_argument(
+    '-u',
+    '--update',
+    action='store_true',
+    default=False,
+    help='Update all rules independent of previous values.')
+parser.add_argument(
+    '-t',
+    '--tighten',
+    action='store_true',
+    default=False,
+    help='Update passing rules if they became tighter')
+parser.add_argument(
+    '-f',
+    '--failing',
+    action='store_true',
+    default=False,
+    help='Update failing rules.')
+args = parser.parse_args()
 
-if len(sys.argv) != 3:
-    print('usage:', sys.argv[0], '<DIR> <VARIANT>')
+if not args.update and not args.tighten and not args.failing:
+    print('[ERROR] Please select at least one of '
+          '-u/--update, -t/--tighten, -f/--failing.')
+    parser.print_help()
     sys.exit(1)
 
-chdir(sys.argv[1])
-variant = sys.argv[2]
-okFile = f'metadata-{variant}-ok.json'
-outFile = f'rules-{variant}.json'
-rules = list()
+chdir(args.dir)
+metrics_file = f"metadata-{args.variant}-ok.json"
+old_rules_file = f"rules-{args.variant}.json"
+new_rules_file = f"rules-{args.variant}.json"
```

Comment:
Why are these the same?  It seems like you are just overwriting and could use the one variable.  Was this for testing?

Reply:
Fixed on [7490572](https://github.com/The-OpenROAD-Project/OpenROAD-flow-scripts/pull/453/commits/74905727591051f695d8319bc6e512781f141e2a)